components/create_comment/create_comment.tsx

```diff
@@ -551,11 +553,20 @@ class CreateComment extends React.PureComponent<Props, State> {
     }
 
     handlePostError = (postError: React.ReactNode) => {
-        this.setState({postError});
+        const state = {postError} as State;
+
+        if (postError) {
+            state.deferringPostSubmit = false;
+        }
+
+        this.setState(state);
```

Comment:
0/5 this can be handled inside the setState call with an updater function:

```
this.setState((state) => ({
    postError,
    deferringPostSubmit: postError ? false : state.deferringPostSubmit,
}))
```

this would reduce the footprint of this handler

Reply:
That's also the recommended way to update state from existing state since it prevents any issues around state not being updated correctly because of how `setState` is batched